tests/Support/Constraint/CssConstraint.php

```diff
@@ -50,20 +49,28 @@ protected static function getCssNeedleRegularExpressionPattern(string $needle):
     {
         $needleMatcher = \preg_replace_callback(
             self::CSS_REGEX_PATTERN,
-            static function (array $matches): string {
-                switch (true) {
-                    case isset($matches[1]) && $matches[1] !== '':
-                        return '\\s*+' . \preg_quote($matches[1], '/') . '\\s*+';
-                    case isset($matches[2]) && $matches[2] !== '':
-                        return '\\s*+';
-                    case isset($matches[3]) && $matches[3] !== '':
-                        return \preg_quote($matches[3], '/') . '\\s*+';
-                    default:
-                        return \preg_quote($matches[0], '/');
-                }
-            },
+            [self::class, 'getCssNeedleRegularExpressionReplacement'],
             $needle
         );
         return '/' . $needleMatcher . '/';
     }
+
+    /**
+     * @param string[] $matches array of matches for {@see CSS_REGEX_PATTERN}
+     *
+     * @return string replacement string, which may be `$matches[0]` if no alteration is needed
+     */
+    private static function getCssNeedleRegularExpressionReplacement(array $matches): string
+    {
+        switch (true) {
```

Comment:
This looks like a misuse of `switch` to me (although it syntactically is valid), and I find it confusing to read. Please let's change this to an `if`/`elseif`/`else` chain that communicates the intent more clearly.

Reply:
OK.  To me it makes it clear, but I appreciate other people's mileage may vary.  Will change.